src/codewhisperer/util/authUtil.ts

```diff
@@ -72,7 +72,17 @@ export class AuthUtil {
                 vscode.commands.executeCommand('aws.codeWhisperer.refreshStatusBar'),
                 vscode.commands.executeCommand('aws.codeWhisperer.updateReferenceLog'),
             ])
-
+            const prompts = PromptSettings.instance
+
+            const shouldShow = await prompts.isPromptEnabled('codeWhispererNewWelcomeMessage')
+            // To check valid connection
+            if (this.isValidEnterpriseSsoInUse() || (this.isBuilderIdInUse() && !this.isConnectionExpired())) {
```

Comment:
shouldn't you check for both if the connection is not expired?

Reply:
isValidEnterpriseSsoInUse() - function already checks the condition. So, i did not write the check again.

```
   public isValidEnterpriseSsoInUse(): boolean {
        return this.isEnterpriseSsoInUse() && !this.isConnectionExpired()
    }
```